### DISCUSSION TITLE
PrimeVuev4 beta 3: Strange behaviour of certain InputText props

### DISCUSSION THREAD

**Discussion by micvm** [primefaces/community#2045] [PrimeVue]
I have a very strange behavior for the `filled` and `disabled` props of the InputText component using Primevue4 (used to work with Primevue3). These two props look to not be reactive somehow. Here an example:

```html
<InputText
  :invalid="isInvalid"
  :pt="{ root: { disabled: isDisabled } }"
  :filled="isFilled"
></InputText>
```

`isInvalid`, `isDisabled` and `isFilled` are reactive boolean data properties. In my application code, `invalid` is updated whenever `isInvalid` changes. As it should. `filled` however, does not change. The same would happen to `disabled` property, so I tried successfully to use passthrough options to pass the reactive `isDisabled` value.

What however bugs me: I tried to code a short reproducer from Stackblitz https://stackblitz.com/edit/6k1d1j-mc9d5y?file=src%2FApp.vue  and, surprisingly, this behavior does not occur there (click on the "toggle" button and then on the input field to check). I tried to include this simple code in my application, isolated from other state, but it does not work there. Any ideas? It works now using passthrough, but I would like to what might be the source of the issue.

**Comment by Niggha2003** [comment]
Have you fixed that yet? I meet the same trouble here and don't know what causes this problem :)))

**Reply by micvm** [reply]
To make it work, you need to pass the properties you need to be reactive as passthrough. The same way as in the above example with the prop `disabled`, you can add the prop `filled`:

```html
<InputText
  :invalid="isInvalid"
  :pt="{ root: { disabled: isDisabled, filled: isFilled } }"
></InputText>
```

 I saw in the source code these two props are handled somehow differently from others, but did not find out why and whether the above "solution" is the way to go.

**Reply by Niggha2003** [reply]
Thank you <3
